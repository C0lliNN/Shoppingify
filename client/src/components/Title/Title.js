import React from 'react';
import styled from 'styled-components';
import * as variables from '../../helpers/style-constants';

const H1 = styled.h1`
  font-family: ${variables.FONT_FAMILY};
  font-size: ${variables.FONT_SIZE_3};
  font-weight: 600;
  @media (min-width: ${variables.LG_BREAK_POINT}px) {
    font-size: ${variables.FONT_SIZE_4};
  }
`;

function Title() {
  return (
    <H1>
      <span style={{ color: variables.COLORS.primary, fontWeight: 'bold' }}>
        Shoppingify{' '}
      </span>
      allows you take your shopping list wherever you go
    </H1>
  );
}

export default Title;
