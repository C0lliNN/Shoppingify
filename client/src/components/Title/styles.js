import * as variables from '../../helpers/style-constants';
import styled from 'styled-components';

export const H1 = styled.h1`
  font-family: ${variables.FONT_FAMILY};
  font-size: ${variables.FONT_SIZE_3};
  font-weight: 600;
  @media (min-width: ${variables.LG_BREAK_POINT}px) {
    font-size: ${variables.FONT_SIZE_4};
  }
`;