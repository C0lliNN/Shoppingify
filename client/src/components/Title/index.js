import React from 'react';
import * as variables from '../../helpers/style-constants';
import { H1 } from './styles';

function Title() {
  return (
    <H1>
      <span style={{ color: variables.COLORS.primary, fontWeight: 'bold' }}>
        Shoppingify{' '}
      </span>
      allows you take your shopping list wherever you go
    </H1>
  );
}

export default Title;
