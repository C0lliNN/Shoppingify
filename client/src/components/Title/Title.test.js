import React from 'react';
import { render } from '@testing-library/react';
import Title from './Title';
import getByTextWithMarkup from '../../helpers/withMarkup';
import 'jest-styled-components';

describe('<Title/>', () => {
  it('should render correctly', () => {
    const { container } = render(<Title />);
    expect(container).toMatchSnapshot();
  });
  it('should display the title text', () => {
    const { getByText } = render(<Title />);

    const element = getByTextWithMarkup(
      getByText,
      'Shoppingify allows you take your shopping list wherever you go'
    );

    expect(element).not.toBeNull();
  });
});
