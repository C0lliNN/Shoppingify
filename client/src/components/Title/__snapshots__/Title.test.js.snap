// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title/> should render correctly 1`] = `
.c0 {
  font-family: Quicksand;
  font-size: 18px;
  font-weight: 600;
}

@media (min-width:1200px) {
  .c0 {
    font-size: 24px;
  }
}

<div>
  <h1
    class="c0"
  >
    <span
      style="color: rgb(249, 161, 9); font-weight: bold;"
    >
      Shoppingify
       
    </span>
    allows you take your shopping list wherever you go
  </h1>
</div>
`;
